modules/security-understanding-container-validation.adoc

```diff
@@ -0,0 +1,157 @@
+
+. The cluster downloads an image manifest and calculates its digest.
+
+. The cluster compares the calculated digest of the image manifest with the known digest.
+If the digest values do not match, the image is rejected without further processing.
+
+. The cluster downloads an image config file, which contains metadata such as SHA256 digests for each part of the container image, and calculates the digest of the image config file.
+
+. The cluster compares the calculated digest of the image config file with the digest that is provided in the image manifest.
+If the digest values do not match, the image is rejected without further processing.
+
+. For each file that comprises the container image, the cluster downloads the file and calculates its digest.
+
+. The cluster compares the calculated digest of each file that makes up the image with the corresponding digest that is provided in the image manifest.
+If the digest values do not match, the image is rejected.
+
+Once the cluster completes all of the previous steps, the integrity of the image has been validated.
+
+[NOTE]
+====
+The image manifest is validated only when the manifest digest is known, which is true in the following cases:
+
+. The image is pulled by specifying the SHA256 image digest, for example `quay.io/openshift-release-dev/ocp-release@sha256:ea7ac3ad42169b39fce07e5e53403a028644810bee9a212e7456074894df40f3`.
+
+. The image signature is also being validated. See “Image authenticity validation” for more information.
+====
+
+[id="security-understanding-image-authenticity_{context}"]
+== Image authenticity validation
+
+In addition to validating the integrity of images, {product-title} clusters also validate their authenticity by certifying that the images were approved by a trusted source, in this case Red{nbsp}Hat.
+This validation process includes checking the digital signature of the images.
+
+Cluster administrators can use `ClusterImagePolicy` resources to declare image policy for the cluster.
+
+:FeatureName: Configuring cluster image policies
+include::snippets/technology-preview.adoc[]
+
+.Example `ClusterImagePolicy` resource 1
+[source,yaml]
+----
+kind: ClusterImagePolicy
+metadata:
+  name: mypolicy-0
+spec:
+  scopes:
+  - test0.com
+  policy:
+    rootOfTrust:
+      policyType: FulcioCAWithRekor
+      fulcioCAWithRekor:
+        fulcioCAData: dGVzdC1jYS1kYXRhLWRhdGE=
+        fulcioSubject: <example_input_here> <1>
+        rekorKeyData: dGVzdC1yZWtvci1rZXktZGF0YQ==
+    fulcioSubject:
+      oidcIssuer: https://OIDC.example.com
+      signedEmail: test-user@example.com
+    signedIdentity:
+      matchPolicy: RemapIdentity
+      remapIdentity:
+        prefix: test-remap-prefix
+        signedPrefix: test-remap-signed-prefix
+
+----
+<1> Specifies OIDC issuer and the email of the Fulcio authentication configuration.
```

Comment:
I think we should also explain the other fields and their possible values.

Reply:
Sure, is there any content that describes these parameters, or could anyone provide me such information?

Reply:
There are Godocs like [these][1].  But I'd expect those to get rendered under [here][2].  Or maybe we only doc GA-cluster APIs there?  For these longer-form docs, I'd expect something that weaves some user stories about how the knobs all come together to deliver a valuable target, with notes to allow customers to riff off that target, and I'm not aware of anything existing today in that space.  For example [this enhancement user story][3] is very thin, and the policy YAML quoted later in the enhancement doesn't have associated text explaining the environment it fits into and how the configuration plugs into that environment.

I agree that these kind of longer stories would be very useful.  I think node team can add them as follow-up improvements on an initial pass in this pull, because y'all are the experts on Sigstore and the sorts of environments customers are likely to set up and want to plug their policies into.

[1]: https://github.com/openshift/api/blob/8f6ded478e8ab589ade8749fb3b70a0fb90a0e60/config/v1alpha1/types_image_policy.go#L181-L195
[2]: https://docs.openshift.com/container-platform/4.16/rest_api/config_apis/config-apis-index.html
[3]: https://github.com/openshift/enhancements/blob/master/enhancements/api-review/add-ClusterImagePolicy-and-ImagePolicy-for-Signature-Verification.md#user-stories

Reply:
Would explaining more of these parameters here honestly be useful in the context of this particular doc? I feel like the purpose of this doc is to mostly just show that ClusterImagePolicy exists and that it now plays a role in internal core image validation.

To me, it seems like if we really wanted to show users how they could use this feature for their own consumption, that would belong in some different doc, perhaps some procedure, and that this other doc would be a better location for any sort of comprehensive reference info.

Reply:
I generally think that there only needs to be one place for full reference documentation, and this article does not look like the canonical place for that. Linking to the reference wouldn’t hurt.